### PR TITLE
Clean up unneeded contig code

### PR DESCRIPTION
**Pull request by gregdingle**
# Description

Following #3157 , we no longer need to compute and send per count taxa info to the report page. 

## Before

image

## After

image

# Tests

* try http://localhost:3000/samples/13435 
* see no change in contig rows before and after

### CODE REVIEW

File: app/services/pipeline_report_service.rb [hunk 1]
```diff
@@ -420,18 +422,14 @@ def merge_contigs(contigs, counts_by_tax_level, csv)
         end
 
         if counts_per_db_type
-          if csv
```
Comment (gregdingle) [inline app/services/pipeline_report_service.rb]: after this change, the non-CSV mode is the same as CSV mode

File: app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx [hunk 2]
```diff
@@ -399,32 +399,21 @@ export default class SampleViewV2 extends React.Component {
     );
   };
 
-  computeRowContigStats = ({ row, minContigReads }) => {
+  setRowContigStats = ({ row, minContigReads }) => {
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx]: You might not need this function.
You probably just need to adjust the key names in the ReportTable: https://github.com/chanzuckerberg/idseq-web/blob/master/app/assets/src/components/views/SampleViewV2/ReportTable.jsx#L156

Comment (tfrcarvalho) [inline app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx]: In any case, `minContigReads` its not used

File: app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx [hunk 3]
```diff
@@ -399,32 +399,21 @@ export default class SampleViewV2 extends React.Component {
...
+      const dbTypeRow = row[dbType];
+      if (dbTypeRow) {
+        dbTypeRow.contigCount = get([dbType, "contigs"], row);
+        dbTypeRow.readsCount = get([dbType, "contig_r"], row);
       }
     });
   };
 
-  computeContigStats = ({ reportData, minContigReads }) => {
+  setContigStats = ({ reportData, minContigReads }) => {
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx]: Should be able to remove this too

**Comment by gregdingle** [comment]
> Ideally we would change get_summary_contig_counts_v2 to aggregate the results in the database, if possible... that function is the true bottleneck...

I don't see an easy way to do this. The current SQL is 
```
SELECT read_count,
       species_taxid_nt,
       species_taxid_nr,
       genus_taxid_nt,
       genus_taxid_nr
FROM `contigs`
WHERE `contigs`.`pipeline_run_id` = 20155
  AND (read_count >= 0)
  AND (lineage_json IS NOT NULL)
```

To aggregate we need a query like this **per count type**
```
SELECT species_taxid_nt,
      "nt",
      SUM(read_count) AS contig_r,
      COUNT(1) AS contigs 
FROM `contigs`
WHERE `contigs`.`pipeline_run_id` = 20155
  AND (read_count >= 0)
  AND (species_taxid_nt IS NOT NULL)
GROUP BY 1
```
I'm not sure if UNIONing 4 such queries would be faster. Do you see another way?

File: app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx [hunk 4]
```diff
@@ -673,11 +639,6 @@ export default class SampleViewV2 extends React.Component {
 
     // different behavior given type of option
     switch (key) {
-      // - min contig size: recompute contig statistics with new size and refresh display
-      case "minContigReads":
```
Comment (gregdingle) [inline app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx]: please confirm this is safe to remove

File: app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx [hunk 5]
```diff
@@ -1106,7 +1067,6 @@ export default class SampleViewV2 extends React.Component {
               reportPresent={!!reportMetadata.reportReady}
               sample={sample}
               view={view}
-              minContigReads={selectedOptions.minContigReads}
```
Comment (gregdingle) [inline app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx]: and this

File: app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx (hunk 2 above)
Comment (gregdingle) [reply app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx]: Thanks for pointing that out. Done.

File: app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx (hunk 4 above)
Comment (tfrcarvalho) [reply app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx]: Yep, sounds safe.

File: app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx (hunk 5 above)
Comment (tfrcarvalho) [reply app/assets/src/components/views/SampleViewV2/SampleViewV2.jsx]: yes. Did you clean up the header on the previous PR?

**Comment by gregdingle** [comment]
Opened https://jira.czi.team/browse/IDSEQ-2420 .

> Did you clean up the header on the previous PR?

I cleaned it up in this PR here: https://github.com/chanzuckerberg/idseq-web/pull/3169/files#diff-6f3d1dcdabd1005da96220e3de664384 . It looks min contigs was only being used there for the CSV download link.

**Pull request merged**